cmd/postgres_exporter/postgres_exporter.go

```diff
@@ -1402,51 +1442,78 @@ func queryNamespaceMappings(ch chan<- prometheus.Metric, server *Server) map[str
 		if scrapeMetric {
 			// Only cache if metric is meaningfully cacheable
 			if mapping.cacheSeconds > 0 {
-				server.cacheMtx.Lock()
+				server.metricCacheMtx.Lock()
 				server.metricCache[namespace] = cachedMetrics{
 					metrics:    metrics,
 					lastScrape: scrapeStart,
 				}
-				server.cacheMtx.Unlock()
+				server.metricCacheMtx.Unlock()
 			}
 		}
 	}
 
 	return namespaceErrors
 }
 
+func cloneValue(src interface{}, dst interface{}) {
+	x := reflect.ValueOf(src)
+	if x.Kind() == reflect.Ptr {
+		starX := x.Elem()
+		y := reflect.New(starX.Type())
+		starY := y.Elem()
+		starY.Set(starX)
+		reflect.ValueOf(dst).Elem().Set(y.Elem())
+	} else {
+		dst = x.Interface() //nolint:staticcheck,ineffassign
```

Comment:
Could you test this func, I does not feel right but I might be wrong. Does Set actually copy values out of server instead of using the maps inside server directly?